### PR TITLE
Remove base config

### PR DESCRIPTION
**Pull request by adameastham**
After going through our linting issues in the consumer app, our list of configurations has grown increasingly away from the base config. I have copied the configurations we do want and removed the base config.

Ill comment on the configurations that have been introduced in this PR too

### CODE REVIEW

File: .swiftlint.yml [hunk 1]
```diff
@@ -1,10 +1,90 @@
...
-# we disable any particularly pedantic rules which do not support auto-correction
 disabled_rules:
+  - discarded_notification_center_observer
+  - notification_center_detachment
+  - orphaned_doc_comment
+  - todo
+  - unused_capture_list
   - attributes
   - indentation_width
+  - function_parameter_count
```
Comment (adameastham) [inline .swiftlint.yml:13]: Added

File: .swiftlint.yml [hunk 2]
```diff
@@ -1,10 +1,90 @@
...
+  max_length:
+    warning: 60
+    error: 80
+  excluded:
+    - i
+    - id
+    - x
+    - y
+    - z
+    - ev
```
Comment (adameastham) [inline .swiftlint.yml:84]: added

File: .swiftlint.yml [hunk 3]
```diff
@@ -1,10 +1,90 @@
...
+
+force_try: warning
+
+function_body_length:
+    warning: 60
+
+legacy_hashing: error
+
+identifier_name:
+  max_length:
```
Comment (adameastham) [inline .swiftlint.yml:75]: added

File: .swiftlint.yml [hunk 4]
```diff
@@ -13,6 +93,19 @@ line_length:
...
+
 trailing_whitespace:
   ignores_empty_lines: true
   ignores_comments: true
+
+vertical_whitespace:
+  max_empty_lines: 2
+
+trailing_closure:
+  only_single_muted_parameter: true
```
Comment (adameastham) [inline .swiftlint.yml:111]: added

**Review by jvanderwee — APPROVED** [review]

File: .swiftlint.yml [hunk 5]
```diff
@@ -1,10 +1,90 @@
...
+  - colon
+  - convenience_type
+  - discouraged_object_literal
+  - empty_collection_literal
+  - empty_count
+  - empty_string
+  - enum_case_associated_values_count
+  - fatal_error_message
+  - first_where
+  - force_unwrapping
```
Comment (bengallard) [inline .swiftlint.yml:33]: Wondering what reason we have for including `force_unwrapping`? I think there's various cases where this should certainly be allowed and imo doesn't really require lint warnings

File: .swiftlint.yml (hunk 2 above)
Comment (bengallard) [reply .swiftlint.yml:84]: Is it possible to only limit max length? Seems a little unreasonable to only allow certain 1-2 char names 🤔

**Review by bengallard — APPROVED** [review]

File: .swiftlint.yml (hunk 5 above)
Comment (adameastham) [reply .swiftlint.yml:33]: @jvanderwee Im up for removing this, there's loads of legit cases in consumer where we force unwrap

Comment (jvanderwee) [reply .swiftlint.yml:33]: it is allowed (albeit with a warning), IMO should be avoided

**Pull request merged**